consensus/common_test.go

```diff
@@ -681,6 +699,24 @@ func ensureVote(voteCh <-chan tmpubsub.Message, height int64, round int32,
 	}
 }
 
+func ensurePrevoteWithNilBlock(voteCh <-chan tmpubsub.Message) {
+	select {
+	case <-time.After(ensureTimeout):
+		panic("Timeout expired while waiting for NewVote event")
+	case msg := <-voteCh:
+		voteEvent, ok := msg.Data().(types.EventDataVote)
+		if !ok {
+			panic(fmt.Sprintf("expected a EventDataVote, got %T. Wrong subscription channel?",
+				msg.Data()))
+		}
+
+		vote := voteEvent.Vote
+		if vote.BlockID.Hash == nil {
```

Comment:
avoid comparing with `nil`, how about check `len(vote.BlockID.Hash)`?

Reply:
do we expect the block ID to be nil or not? the function says *ensure it's nil* but here says if it's nil it should panic

Reply:
it is expected to be nil. the panic is triggered on purpose for checking panic msg in unit test